runtime/runtime/src/config.rs

```diff
@@ -12,46 +12,104 @@ pub struct RuntimeConfig {
     /// The minimum number of blocks of storage rent an account has to maintain to prevent forced deletion.
     pub poke_threshold: BlockIndex,
     /// Costs for different types of transactions.
-    pub transactions_costs: TransactionsCosts,
+    pub transaction_costs: TransactionCosts,
     /// Config of wasm operations.
     pub wasm_config: Config,
 }
 
 /// The costs of the transactions.
 #[derive(Default, Debug, Serialize, Deserialize, Clone)]
-pub struct TransactionsCosts {
-    pub create_account: Balance,
-    pub deploy_contract: Balance,
-    pub function_call: Balance,
-    pub self_function_call: Balance,
-    pub send_money: Balance,
-    pub stake: Balance,
-    pub swap_key: Balance,
-    pub add_key: Balance,
-    pub delete_key: Balance,
-    pub delete_account: Balance,
+pub struct TransactionCosts {
```

Comment:
This structure might need to go entirely into `near-vm-logic`.

Reply:
So this is what I start to wonder. If we decide to charge for every byte of the message (e.g. for args over network). This logic might need to be replicated with the VM, since both of them should produce same Gas cost results.

Reply:
I suggest we do the following:
For each binding function of VM we define a cost function: `F(args, result)->Gas`. Which represents the fees that will be applied to call this function, including side-effects like creating receipts. We have this function in `near-vm-logic` and the `runtime` crate receives this information as part of `used_gas`.

Pros:
* The logic is encapsulated in `near-vm-logic` which means it can be used for unit tests in smart contracts. People can write a smart contract, cover it with unit tests and if they did good job with coverage they can expect it to work 100% on the blockchain, because anything not covered by `near-vm-*` is guaranteed by the rest of engine (like receipts delivery, etc).
* Our economics-related stuff is in one place, so we do a good job testing it to prevent all kinds of attacks, secondary markets, etc.

Cons:
* `F(args, result)->Gas` will not represent the actual cost of sending a receipt over the network, including serialization etc. However, independently on our architecture we cannot devise a function (whether it is in `runtime` or `near-vm-logic`) that can encapsulate all side-effects and costs, so it will always be an approximation. For instance we do not account for temporary storage that cross-contract calls create in the trie because we cannot measure how many nodes deep it will be, etc. So I suggest we always think of this fee as an approximation, and since it is an approximation, why not keep it in `near-vm-logic`.